draft-ietf-privacypass-auth-scheme-extensions.md

```diff
@@ -123,17 +127,63 @@ issuance protocol. In particular, the value of an Extensions structure is provid
 metadata for the issuance protocol. Candidate issuance protocols are specified in
 {{?PUBLIC-ISSUANCE=I-D.hendrickson-privacypass-public-metadata}}.
 
-# Extensions Negotiation {#negotiation}
+# PrivateToken Challenge Extension Parameter
+
+As defined in {{Section 2.1 of AUTHSCHEME}}, the "PrivateToken" authentication
+scheme defines two parameters, "challenge" which contains the base64url-encoded
+TokenChallenge struct, and a "token-key" which contains the base64url-encoded
+public key used for this challenge. This document defines two OPTIONAL new parameters, 
+"extension-set," which contains the base64url-encoded representation of the
+following ExtensionSet structure, and "extensions" which contain the base64url-encoded 
+representation of the Extensions strucuture defined in {#extensions}. This document 
+follows the default padding behavior described in {{Section 3.2 of !RFC4648}}, so the
+base64url value MUST include padding.
+
+~~~
+struct {
+    enum { false(0), true(1) } Bool;
+    Bool is_required;
+    ExtensionType extension_type;
+} ExtensionEntry;
+
+enum {
+    reserved(0),
+    (65535)
+} ExtensionType;
+
+struct {
+    ExtensionEntry extension_types<0..2^16-1>;
+} ExtensionSet;
+~~~
+
+The contents of ExtensionSet is a list of ExtensionEntry structs containing extensions (defined in #extensions),
```

Comment:
I don't see Extensions struct defined as a member of ExtensionEntry struct.

Reply:
@pbsawant, would it make sense to signal the entire extension in the challenge? The extension types should be sufficient for extension negotiation.

Reply:
Agreed I don't think we need to define Extension within ExtensionEntry, as the Entries are just meant to signal which ExtensionType is is supported, along with required-ness.